src/redux/missions/missions.js

```diff
@@ -1,14 +1,65 @@
+/* eslint-disable no-case-declarations */
 import axios from 'axios';
 
 const GETMISSIONS = 'app/missions/MISSIONS_RETRIEVED';
+const JOINMISSION = 'app/missions/MISSION_JOINED';
+const LEAVEMISSION = 'app/missions/MISSION_LEFT';
 
-const reducer = (missions = [], action) => {
+const reducer = (state = [], action) => {
   const { type, payload } = action;
   switch (type) {
     case GETMISSIONS:
+      let missionsSaved;
+      if (localStorage.getItem('missions')) {
+        missionsSaved = JSON.parse(localStorage.getItem('missions'));
+        return [...missionsSaved];
+      }
+      localStorage.setItem('missions', JSON.stringify(payload));
```

Comment:
This is a good idea, but we should not have a side effect in the reducer.
Plus, it is strictly said that we should not persist user data in the project requirements.

> __Technical set up**__
>  - Set up the repository on GitHub and use Gitflow.
>  - Set up React and Redux.
>  - Set up testing libraries(React Testing Library and Jest).
>  - Do not preserve user data.

Kindly check it [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md) near the bottom of the page.

Reply:
@jssol  alright thanks for pointing it out, will update accordingly

Reply:
Great

Reply:
@amiraabouhadid, By the way, I'm still awaiting your review in this [PR](https://github.com/amiraabouhadid/space_travelers_hub/pull/28)

Reply:
@jssol hey yeah I reviewed it already

Reply:
@jssol I have implemented the changes you requested, please re-review and let m know if I need to make more changes.
Thanks, 
Amira